### DISCUSSION TITLE
How can I install all LSPs with one command?

### DISCUSSION THREAD

**Discussion by Andy3153** [williamboman/nvim-lsp-installer#647] [Q&A] [answered]
I've seen there is a `:LspUninstallAll`, but no `:LspInstallAll`, so, how can I install all of them with one command?

**Comment by williamboman** [comment, marked as answer]
Hello! As of writing there are ~115 servers available for installation. It doesn't make much sense to install all of these because

- It would carry an epic overhead, in terms of 1) the resources required to install them, and 2) the footprint it would leave on the file system
- Many languages have multiple different server implementations, for example python has 3+ servers. Using more than one per language is an extremely rare use case.

**Reply by williamboman** [reply]
There exists an option to automatically install the servers that you actually set up, which might solve your use case:

```lua
require("nvim-lsp-installer").setup {
  automatic_installation = true
}
```

**Reply by Andy3153** [reply]
I know about the overhead, storage is not a problem for me and I can bear with the installation time. Plus, it's done only once

The reason I want to install them all is to have a more "set it and forget it" experience, and not need to remember to install a new LSP every time I edit a file with a different filetype

**Reply by williamboman** [reply]
You could make use of the Lua APIs documented in `:h nvim-lsp-installer` and code a small snippet to install all available servers automatically. I won't write out an example here because I don't want people copy pasting it in their own configs for the reasons mentioned above!
